### ISSUE TITLE
uni.scanCode 可以在 success 回调里对外部变量赋值，uni.chooseImage 不可以

### ISSUE THREAD

**Issue by gyjcode** [dcloudio/uni-app#4620]
**问题描述**
uni.scanCode 可以在 success 回调里对外部变量赋值，uni.chooseImage 不可以。这是为什么啊？

**复现步骤**
let result: any
  uni.chooseImage({
    count: 1,
    sizeType: ['original', 'compressed'],
    sourceType: ['album'],
    success: (res: any ) => {
      result = '123';
    }
  });
  console.log('scanFileH5', result)
  uni.scanCode({
    success: (res) => {
      result = '456';
    }
  });
  console.log('scanCodeApp', result)

**预期结果**
scanFileH5 123
scan.ts:178 scanCodeApp 789

**实际结果**
scanFileH5 undefined
scan.ts:178 scanCodeApp 789

**系统信息:**
 Window 10 npm run dev:h5 测试

**Comment by gyjcode** [comment]
> api是异步的
scanCode是同步的，chooseImage 是异步的吗？

**Comment by 18148764734** [comment]
是的。测试了一下，确实是同步异步的问题，无论是scanCode 还是 chooseImage都是异步的，你把console.log()执行在sucess里面就知道了，都是已经赋值成功了的

**Comment by 18148764734** [comment]
请问你是VUE2还是VUE3呢？是运行在哪个端才会复现这个问题呢？能否提供问题工程？

**Comment by GRCmade** [comment]
长时间未回复，关闭 issue，如果还存在这个问题，可以重新打开
